### ISSUE TITLE
About the lrp and group conv

### ISSUE THREAD

**Issue by YuKDseele** [qingshi9974/ICLR2024-FTIC#9]
Hello, I have trouble understanding the `TCA_EntropyModel` section of the code and hope to get some clarification.

```python
self.entropy_parameters_net = nn.Sequential(
    nn.Conv2d(dim * (self.ratio + 2), dim * self.ratio // 2, 3, 1, 1, groups=self.slices),
    nn.GELU(),
    nn.Conv2d(dim * self.ratio // 2, dim * 3, 3, 1, 1, groups=self.slices),
    nn.GELU(),
    nn.Conv2d(dim * 3, dim * 3, 3, 1, 1, groups=self.slices),
)
```

In the code, it seems that the network simultaneously outputs **mean**, **scale**, and **lrp**. This differs from earlier papers, which usually normalize \(y\) using **mean** and **scale**, quantize \(y\) with a quantizer, and then explicitly use an LRP network. For example:

```python
y_hat_slice = quantize_ste(y_slice - mu) + mu
if slice_index > 0:
    lrp_support = torch.cat([mu, torch.cat(y_hat_slices, dim=1)], dim=1)
else:
    lrp_support = mu
lrp = self.lrp_transforms[slice_index](lrp_support)
lrp = 0.5 * torch.tanh(lrp)
y_hat_slice += lrp
y_hat_slices.append(y_hat_slice)
```

However, in the provided implementation:

```python
out = self.entropy_parameters_net(torch.cat((out1, hyper1), 2).view(B, C * (self.ratio + 2), H, W)).view(B, C, 3, H, W).contiguous()

means = out[:, :, 0].contiguous()
scales = out[:, :, 1].contiguous()
lrp = out[:, :, 2].contiguous()
```

I would like to know why the it this way, where **mean**, **scale**, and **lrp** are directly predicted together, instead of following the sequential steps used in earlier works.

Besides, this paper mentions that group convolution improves the modeling of dependencies within slices, and \(ng\) is set to the number of slices \(ns\). I would like to know if there is a specific rationale for this, and whether \(ng\) can be set to other values?

**Comment by qingshi9974** [comment]
please refer to https://github.com/qingshi9974/ICLR2024-FTIC/issues/4#issuecomment-2206543721
